src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -68,6 +70,14 @@ class AuthScreens extends React.Component {
             authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
         }).then(subscribeToReportCommentEvents);
 
+        // If we have a validation code in our URL ensure we validate the associated login
+        if (this.props.currentURL.includes(`/${ROUTES.VALIDATE_LOGIN}/`)) {
+            // Split the /v/accountID/validateCode URL string into distinct components
+            // "", "v", "accountID", "validateCode"
+            const urlArray = this.props.currentURL.split('/');
+            User.validateLogin(Number(urlArray[2]), urlArray[3]);
```

Comment:
It's very possible that the code `Number(urlArray[2]), urlArray[3]` could break things if the URL is not exactly what it's expected to be. I'd suggest using something more like `lodash.get` to access the accountID and code.

Reply:
ah yes, good point here.